docs/specification/draft/client/elicitation.mdx

```diff
@@ -317,16 +521,199 @@ The three response actions are:
+```
+
+This pattern maintains clear security boundaries while enabling rich integrations with third-party services that require user authorization.
+
 ## Security Considerations
 
-1. Servers **MUST NOT** request sensitive information through elicitation
+1. Clients **MUST** provide clear indication of which server is requesting information
 2. Clients **SHOULD** implement user approval controls
-3. Both parties **SHOULD** validate elicitation content against the provided schema
-4. Clients **SHOULD** provide clear indication of which server is requesting information
-5. Clients **SHOULD** allow users to reject elicitation requests at any time
-6. Clients **SHOULD** implement rate limiting
-7. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+3. Clients **SHOULD** allow users to reject elicitation requests at any time
+4. Clients **SHOULD** implement rate limiting
+5. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+6. Servers **MUST** bind elicitation requests to the user's identity
+
+### Safe URL Handling
+
+Clients implementing out-of-band elicitation **MUST** handle URLs carefully to prevent users from unknowingly clicking malicious links.
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
+2. Servers **MUST NOT** include URLs in any message or schema fields as part of an out-of-band elicitation request, except for the `url` field.
+
+These requirements ensure that client implementations have clear rules about when to present a URL to the user, so that other rules about user consent and SSRF protection (below) can be consistently applied.
+
+### Identifying the User
+
+Servers **MUST NOT** rely on client-provided user identification, as this can be forged.
+Instead, servers **SHOULD** follow [security best practices](/specification/draft/basic/security_best_practices).
+
+Non-normative examples:
+
+- Incorrect: Treat user input like "I am joe@example.com" as authoritative
+- Correct: Rely on the [MCP authorization server](/specification/draft/basic/authorization) to identify the user
+
+### Form Mode Security
+
+1. Servers **MUST NOT** request sensitive information (passwords, API keys, etc.) via form mode
+2. Clients **SHOULD** validate all responses against the provided schema
+3. Servers **SHOULD** validate received data matches the requested schema
+
+### Out-of-Band Mode Security
+
+Clients **MUST NOT** open a user agent (e.g. browser) without explicit consent from the user
+
+#### Server-Side Request Forgery (SSRF)
+
+Since clients facilitate the opening of URLs provided by servers, they **MUST** implement SSRF protections, including:
+
+- Requiring the `https://` scheme for all out-of-band URLs (no HTTP, `file://`, etc.)
+- Blocking requests to internal IP ranges (e.g., `127.0.0.1`, `10.0.0.0/8`, etc.)
```

Comment:
stdio servers eliciting OOB on localhost servers is going to be super common

Reply:
Any scenario where they want a rich UI or interaction versus the simple `form` based Elicitation. Random creative example: I have an image editor MCP and want to give the user the option to pop up a rich image editor GUI before I upload the new design version to the cloud.